StandardLibraryEssentials/StandardTemplateLibrary/STLAlgorithms/task.md

```diff
@@ -0,0 +1,179 @@
+```cpp
+std::max_element(start_iterator, end_iterator, comparator);
+std::min_element(start_iterator, end_iterator, comparator);
+```
+The first two arguments are iterators that define the range of elements to be searched in. The third argument is an optional comparator used to compare two elements. If no comparator is provided, the elements are compared using the `<` operator by default.
+```cpp
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+int main() {
+    std::vector<int> v = {1, 2, 3, 4, 5};
+    auto max_it = std::max_element(v.begin(), v.end());
+    auto min_it = std::min_element(v.begin(), v.end());
+    std::cout << "Max: " << *max_it << std::endl;
+    std::cout << "Min: " << *min_it << std::endl;
+}
+```
+
+#### 5. `std::fill()`
+[`std::fill()`](https://en.cppreference.com/w/cpp/algorithm/fill) is used to fill a range of elements with a value. it is iterated in the time it takes to traverse the entire array to change the elements `O(n)`. The syntax is quite simple:
+```cpp
+std::fill(start_iterator, end_iterator, value);
+```
+The first two arguments are iterators that define the range of elements to be filled. The third argument is the value to be filled.
+```cpp
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+int main() {
+    std::vector<int> v = {1, 2, 3, 4, 5};
+    std::fill(v.begin(), v.end(), 0);
+    // v = {0, 0, 0, 0, 0}
+}
+```
+
+Also, there is a header called [`<numeric>`](https://en.cppreference.com/w/cpp/header/numeric) that contains a number of useful algorithms for numeric operations.
+
+#### 1. `std::accumulate()`
+[`std::accumulate()`](https://en.cppreference.com/w/cpp/algorithm/accumulate) is used to calculate the sum of a range of elements. It is implemented as a simple linear `O(n)` traversal, with this syntax:
+```cpp
+std::accumulate(start_iterator, end_iterator, initial_value);
+```
+The first two arguments are iterators that define the range to be summed. The third argument is the initial value of the sum. Be aware of the accumulator type, as it may overflow if the sum is too large. For example, if you want to sum a range of integers, you can use `0LL` as the initial value to avoid overflow.
+```cpp
+#include <iostream>
+#include <vector>
+#include <numeric>
+
+int main() {
+    std::vector<int> v = {1, 2, 3, 4, 5, 6, 7, 8, 9};
+    long long sum = std::accumulate(v.begin(), v.end(), 0LL);
+    std::cout << "Sum: " << sum << std::endl;
+}
+```
+
+#### 2. `std::gcd()` and `std::lcm()`
+[`std::gcd()`](https://en.cppreference.com/w/cpp/numeric/gcd) and [`std::lcm()`](https://en.cppreference.com/w/cpp/numeric/lcm) are used to calculate the Greatest Common Divisor and the Least Common Multiple of two integers. They are implemented as `O(log n)` algorithms, looking like that:
```

Comment:
You would not encounter `gcd` and `lcm` often in your code, so let's avoid mentioning them.